src/components/Sections/SectionActions.tsx

```diff
@@ -1,35 +1,122 @@
+import RadioChecked from 'cozy-ui/transpiled/react/Icons/RadioChecked'
+import RadioUnchecked from 'cozy-ui/transpiled/react/Icons/RadioUnchecked'
+import Switch from 'cozy-ui/transpiled/react/Switch'
+import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import {
+  Action,
+  divider,
+  makeActions
+} from 'cozy-ui/transpiled/react/ActionsMenu/Actions'
+import { useSettings } from 'cozy-client'
+
+import {
+  DisplayMode,
+  GroupMode,
+  Section,
+  SectionViewProps
+} from 'components/Sections/SectionsTypes'
+import {
+  computeDisplayMode,
+  handleSectionAction
+} from 'components/Sections/utils'
+
+const isDisplayMode = (
+  value: DisplayMode | GroupMode
+): value is DisplayMode => {
+  return Object.values(DisplayMode).includes(value as DisplayMode)
+}
+
+// This feature is disabled in this pull request
+const StateSwitch = (): JSX.Element => {
+  return <Switch className="u-w-auto u-pr-0" disabled />
+}
 
-const createSectionAction = (primaryText: DisplayMode): (() => Action) => {
-  return () => ({
-    name: `sectionAction_${primaryText}`,
-    action: (_doc, opts): void => {
-      opts.handleAction(primaryText)
-    },
-    Component: React.forwardRef(function SectionActionComponent(props, ref) {
-      return (
-        <ActionsMenuItem {...props} ref={ref}>
-          <ListItemIcon>
-            <Icon icon={FileIcon} />
-          </ListItemIcon>
-          <ListItemText primary={primaryText} />
-        </ActionsMenuItem>
+const createSectionAction = (
+  actionLabel: DisplayMode | GroupMode
+): (() => Action) => {
+  if (isDisplayMode(actionLabel)) {
+    return () => ({
+      name: `sectionAction_${actionLabel}`,
+      Component: React.forwardRef<{ docs?: Section[] }>(
+        function SectionActionComponent(props: { docs?: Section[] }, ref) {
+          const { isMobile } = useBreakpoints()
+          const { values, save } = useSettings('home', ['shortcutsLayout'])
+          const section = props.docs?.[0] as SectionViewProps['section']
+          const currentDisplayMode = computeDisplayMode(isMobile, section)
+          const handleClick = (): void =>
+            handleSectionAction(section, isMobile, actionLabel, values, save)
+          const { t } = useI18n()
+          const isActive = actionLabel === currentDisplayMode
+
+          return (
+            <ActionsMenuItem
+              {...props}
+              ref={ref}
+              onClick={handleClick}
+              className="u-miw-auto u-w-auto"
+            >
+              <ListItemIcon>
+                <Icon
+                  icon={
+                    actionLabel === DisplayMode.COMPACT ? MosaicMin : ListMin
+                  }
+                />
+              </ListItemIcon>
+
+              <ListItemText
+                className="u-mr-half"
+                primary={t(`sections.label_${actionLabel}`)}
+              />
+
+              <Icon
```

Comment:
You can use a radio directly here make the button appear like in Store ([example](https://github.com/cozy/cozy-store/blob/e1693f040c21e1a1db342860ff6a764bccc02711/src/ducks/apps/components/Sections/components/Filters.jsx#L48))

Reply:
Updated ✅